ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java

```diff
@@ -25,15 +25,21 @@
 @JsonSubTypes({})
 public class Command {
   private final String statement;
+  private final int commandTopicOffset;
   private final Map<String, Object> overwriteProperties;
   private final Map<String, String> originalProperties;
   private final boolean preVersion5;
 
+  public static final Integer DEFAULT_COMMAND_TOPIC_OFFSET = -1;
+
   @JsonCreator
   public Command(@JsonProperty("statement") final String statement,
+                 @JsonProperty("commandTopicOffset") final Integer commandTopicOffset,
                  @JsonProperty("streamsProperties") final Map<String, Object> overwriteProperties,
                  @JsonProperty("originalProperties") final Map<String, String> originalProperties) {
     this.statement = statement;
+    this.commandTopicOffset =
```

Comment:
I personally like offset because it also provides some guarantees its characteristics (i.e. monotonically increasing) - so it gives more, perhaps useful, information beyond "signature"